### PR TITLE
gnrc_ipv6_static_addr: fix build with only static address

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Using `gnrc_ipv6_static_addr` with only `CONFIG_GNRC_IPV6_STATIC_ADDR` set fails due to a copy & paste error.

Also remove the need to specify the interface if there is only one.

### Testing procedure

Running `examples/gnrc_networking` with 

```make
USEMODULE += gnrc_ipv6_static_addr
CFLAGS += -DCONFIG_GNRC_IPV6_STATIC_ADDR=\"2001:db8::2\"
```

now has the expected result

```
Iface  7  HWaddr: 4F:C5  Channel: 26  NID: 0x23  PHY: O-QPSK 
          Long HWaddr: A6:8F:3A:12:5E:63:4F:C5 
           State: IDLE 
          ACK_REQ  L2-PDU:102  MTU:1280  HL:64  RTR  
          6LO  IPHC  
          Source address length: 8
          Link type: wireless
          inet6 addr: fe80::a48f:3a12:5e63:4fc5  scope: link  VAL
          inet6 addr: 2001:db8::2  scope: global  VAL
          inet6 group: ff02::2
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:4fc5
          inet6 group: ff02::1a
          inet6 group: ff02::1:ff00:2
```

### CODE REVIEW

File: sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c [hunk 1]
```diff
@@ -117,6 +117,10 @@ void auto_init_gnrc_ipv6_static_addr(void)
     gnrc_netif_t *upstream = NULL;
     gnrc_netif_t *downstream = NULL;
 
+    if (gnrc_netif_numof() == 1) {
+        upstream = gnrc_netif_iter(NULL);
+    }
+
```
Comment (miri64) [inline sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c:123]: Use `gnrc_netif_highlander()` here, to optimize here, too, so all the searching below can be optimized away?
```suggestion
    if (gnrc_netif_highlander() || gnrc_netif_numof() == 1) {
        upstream = gnrc_netif_iter(NULL);
        /* ... */
    }
    else {
        /* ... */
    }
```

**Review by miri64 — CHANGES_REQUESTED** [review]

File: sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c (hunk 1 above)
Comment (benpicco) [reply sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c:123]: I somehow thought that was already part of `gnrc_netif_numof()`

File: sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c [hunk 2]
```diff
@@ -117,6 +117,10 @@ void auto_init_gnrc_ipv6_static_addr(void)
     gnrc_netif_t *upstream = NULL;
     gnrc_netif_t *downstream = NULL;
 
+    if (gnrc_netif_highlander() || gnrc_netif_numof() == 1) {
+        upstream = gnrc_netif_iter(NULL);
+    }
+
```
Comment (miri64) [inline sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c:123]: There is still an else needed for the while loop not to be compiled in. Or does an early return here make sense?

Comment (benpicco) [reply sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c]: Ah good point

File: sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c [hunk 3]
```diff
@@ -113,25 +113,30 @@ static void _config_downstream(gnrc_netif_t *downstream)
...
 {
-    gnrc_netif_t *netif = NULL;
     gnrc_netif_t *upstream = NULL;
     gnrc_netif_t *downstream = NULL;
 
-    if (IS_ACTIVE(CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM)) {
-        upstream = gnrc_netif_get_by_pid(CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM);
-    }
+    if (gnrc_netif_highlander() || gnrc_netif_numof() == 1) {
+        upstream = gnrc_netif_iter(NULL);
```
Comment (miri64) [inline sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c:120]: Would it make sense to at least issue a warning that this is might be `!= CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM` (i.e. effectively ignoring a potentially nonsensical configuration)? Or even have an assertion here?
```suggestion
        upstream = gnrc_netif_iter(NULL);
        if (IS_ACTIVE(CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM) &&
            (CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM != upstream->pid) {
            LOG_WARNING("Only netif %d differs from selected upstream netif %d\n", netif->pid,
                        CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM);
        }
```

Or

```suggestion
        upstream = gnrc_netif_iter(NULL);
        assert(!IS_ACTIVE(CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM) ||
               CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM == upstream);
```

Comment (benpicco) [reply sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c:120]: Well if you insist…
I really think this makes the code more complicated than it needs to be

Comment (miri64) [reply sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c:120]: Does it? In any case it provides a user that might have misconfigured their system with valuable information why things don't work the way they expect it to be IMHO. The less complicated solution is the assert if you ask me.

Comment (miri64) [reply sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c:120]: Thanks!

**Review by miri64 — APPROVED** [review]

**Comment by benpicco** [comment]
bors merge

**Comment by benpicco** [comment]
bors cancel
bors merge

**Pull request merged**